src/js/yeldaWebchatInjector.js

```diff
@@ -949,6 +965,48 @@ class YeldaChat {
       webchatFrame.contentWindow.postMessage({ event: 'sendUserMessage', data: message })
     }
   }
+
+  /**
+   * Listen leave viewport event handling
+   * @param {Boolean} [remove=false]
+   */
+  listenLeaveViewport(remove = false) {
+    const md = new MobileDetect(navigator.userAgent)
+    const isMobile = md.mobile() !== null
+
+    // Not needed to listen on mobile
+    if (isMobile) {
+      return
+    }
+
+    const eventHandler = window[remove ? 'removeEventListener' : 'addEventListener']
+
+    if (!this.viewportListenerBind) {
+      this.viewportListenerBind = this.viewportListener.bind(this)
+    }
+    // mouseout event is used to listen leave viewport
+    eventHandler('mouseout', this.viewportListenerBind)
+
+    if (remove) {
+      this.pushNotificationId = null
+    }
+  }
+
+  /**
+   * Callback for leave viewport listener
+   * @param {Event} event
+   */
+  viewportListener(event) {
+    if (event.toElement || event.relatedTarget) {
+      return
+    }
```

Comment:
Maybe add a comment explaining what these events are and why we are ignoring them?